assembly/src/tests.rs

```diff
@@ -2486,3 +2486,60 @@ fn test_compiled_library() {
 
     let _program = assembler.assemble_program(program_source).unwrap();
 }
+
+#[test]
+fn test_reexported_proc_with_same_name_as_local_proc_diff_locals() {
+    let context = TestContext::new();
+    let mut mod_parser = ModuleParser::new(ModuleKind::Library);
+    let mod1 = {
+        let source = source_file!(
+            &context,
+            "export.foo.2
+    push.1
+    drop
+end
+"
```

Comment:
nit: formatting is a bit off here.

Reply:
This is intentional, so that diagnostics are rendered properly, you aren't supposed to indent multi-line strings, as the leading whitespace is preserved in the string. I didn't want to fix this up globally for this PR, but we should do that eventually.